### ISSUE TITLE
facenet.mlmodel output 512 length vector, is there a model that output 128

### ISSUE THREAD

**Issue by Nomia** [hosituanit/clockon-clockoff-face-recognition#1]
hi, thanks for this project

I noticed that your article in the `medium.com` provided a download link of `facenet.mlmodel`, but it will output `512` length vector,  is there a model that output `128` length vector?

**Comment by Nomia** [comment]
I got a `openface.mlmodel` that produce a 128 element vector, it's more accurate than the `facenet`(512 element vector), but that openface.mlmodel cannot work on iphone12

**Comment by hosituan** [comment]
I updated readme with this model https://drive.google.com/file/d/1WjZ-rX-Jq0U08U5VefQ7XSC5Dc6RPitI/view 
You can download this model and put it in Graph folder

**Comment by hosituan** [comment]
Or you can use .mlmodel and set up manualy similar to my approach in medium post

**Comment by Nomia** [comment]
thanks, it works now

I was wondering why do u use the `cidetector` to detect faces instead of using the `Vision API` by Apple to detect faces, specifically the `VNDetectFaceLandmarksRequest` method

**Comment by hosituan** [comment]
I tried it, but actually I don't know why it does not accurate than `cidetector`, you can try to use it. But remember that you need to use the same approach when you get face data and recognize face

**Comment by Nomia** [comment]
I tried that with `openface.mlmodel`, 96 * 96 aligned face input, 128 output, works pretty well as for face recognition, but now I try to use `Vision API` with the `facenet.pb`, 160 * 160 aligned face input, 128 output, accuracy is much lower, I'll try the `cidetector` to implement the function to see what will happen

and, I wanna give you a reference as for the comparison between the face models, hopefully it will be helpful to you
![image](https://user-images.githubusercontent.com/1707818/130596293-30d4b3f9-7bea-4e4a-98e1-f505fef88796.png)

here is the video: https://www.youtube.com/watch?v=i_MOwvhbLdI&ab_channel=SefikIlkinSerengil

**Comment by Nomia** [comment]
by the way, could u tell me what training data was the `modelFacenet.pb` trained on

thanks

**Comment by hosituan** [comment]
It is pretrained model from this repo, convert to `.pb` https://github.com/davidsandberg/facenet

**Comment by Nomia** [comment]
ok, got it
I just tried combined the `cidetector` with `modelFacenet.pb` for face recognition, works better than the combination of `Vision API` and `OpenFace.mlmodel`
